internal/resources/planner.go

```diff
@@ -301,7 +301,7 @@ func (sp *sharePlanner) update() (changed bool, err error) {
 	return
 }
 
-func (sp *sharePlanner) prune() (changed bool, err error) {
+func (sp *sharePlanner) Prune() (changed bool, err error) {
```

Comment:
I think we can probably remove this function as it probably is really unused now.

Reply:
Would this function be needed in the future? If so, maybe it is better to leave it as is and just add "// nolint:unused" .

Reply:
I'm fairly certain that it's now unused as I abandoned the earlier appraoch of trying to use a shared config map across multiple instances. However, don't feel obligated to clean that up now as long as the function is left private, I'd be fine with adding the special comment. We can double check that it's unused and remove it in a future pr.